### DISCUSSION TITLE
Can't get automerge to work

### DISCUSSION THREAD

**Discussion by dave-yotta** [renovatebot/renovate#25167] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate PRs are merged when required checks pass automatically.

### What you tried so far.

I have enabled automerge with package rule on a couple of repos, I will leave config at bottom.  Renovate is installed via github app.  App has read and write permissions to code and opens PRs.  One repo (alloy-reports) shows PR with automerge enabled, PR for other (alloy-engine) does not.  I have enabled automerge in settings for the repo, and added branch protection to the base branch adding required check only.  The PRs were already open before making the changes to renovate.json - both updated to reflect new baseBranch, but only one changed to reflect automerge enabled.

Not sure what's missing, can't find anything in the logs or anything in the troubleshooting automerges as to what to look for in the logs.

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "dependencyDashboard": true,
  "prHourlyLimit": 1,
  "prConcurrentLimit": 1,
  "baseBranches": ["renovate/dev"],
  "rebaseWhen": "never",
  "packageRules":
  [
    {
      "matchBaseBranches": ["renovate/dev"],
      "automerge": true
    }
  ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>
alloy-reports: https://gist.github.com/dave-yotta/ddcf723e495e047d0aac56ff5f425f21
</details>

**Comment by rarkins** [comment]
Pick one branch/PR to focus on. Which one?

**Reply by dave-yotta** [reply]
Ok - lets look at alloy-reports.  This is the one which has PRs indicating 🚦 Automerge: Enabled.  I've updated the description to point to a gist for the latest run on the alloy-reports repo.

**Reply by rarkins** [reply]
alloy-reports is the name of the repository, not a dependency/branch/PR. I meant I want you to pick one PR within the log you shared, because it's long

**Reply by dave-yotta** [reply]
Ok gotcha.  Lets do:
```
alloy-reports/renovate/node-17.x-lockfile
chore(deps): update dependency @types/node to v17.0.45 #263
```

**Reply by rarkins** [reply]
`DEBUG: User has requested rebase (branch="renovate/node-17.x-lockfile")`

**Reply by dave-yotta** [reply]
That was two hours ago - I was trying a re-run to get automerge working.  It completed an hour ago:

![image](https://github.com/renovatebot/renovate/assets/41444963/2722fa46-5d15-4a9f-a3a1-0fe3417ce922)

**Reply by rarkins** [reply]
Asking a branch to retry/rebase is not the way to get it to automerge - it forces a new commit.

The hosted app runs a scheduled job every 3-4 hours per repo, so it's no surprise if it hasn't run for 2 hours since the last commit.

You can use the dependency dashboard and tick the `Check this box to trigger a request for Renovate to run again on this repository` checkbox to trigger a new job for the repo.

**Reply by dave-yotta** [reply]
I've triggered a run, still not merging:
![image](https://github.com/renovatebot/renovate/assets/41444963/f8e636cc-440b-4207-95d8-7e6b8818139b)
logs for that run: https://gist.github.com/dave-yotta/293ed84f7d2f956c70cef1bdec4a7571

**Reply by rarkins** [reply]
What does the log say for that particular branch?

**Reply by dave-yotta** [reply]
Grepping the above for the branch name gives these debug lines:
```
DEBUG: getBranchPr(renovate/node-17.x-lockfile)
DEBUG: findPr(renovate/node-17.x-lockfile, undefined, open)
DEBUG: 3 already existing branches found: renovate/less-3.x-lockfile,renovate/node-17.x-lockfile,renovate/glob-7.x-lockfile
DEBUG: syncBranchState() (branch="renovate/node-17.x-lockfile")
DEBUG: branch.isUpToDate(): using cached result "true" (branch="renovate/node-17.x-lockfile")
DEBUG: getBranchPr(renovate/node-17.x-lockfile) (branch="renovate/node-17.x-lockfile")
DEBUG: findPr(renovate/node-17.x-lockfile, undefined, open) (branch="renovate/node-17.x-lockfile")
DEBUG: Found PR #263 (branch="renovate/node-17.x-lockfile")
DEBUG: branchExists=true (branch="renovate/node-17.x-lockfile")
DEBUG: dependencyDashboardCheck=undefined (branch="renovate/node-17.x-lockfile")
DEBUG: PR rebase requested=false (branch="renovate/node-17.x-lockfile")
DEBUG: Check for closed PR because recreating closed PRs is disabled. (branch="renovate/node-17.x-lockfile")
DEBUG: findPr(renovate/node-17.x-lockfile, chore(deps): update dependency @types/node to v17.0.45, !open) (branch="renovate/node-17.x-lockfile")
DEBUG: prAlreadyExisted=false (branch="renovate/node-17.x-lockfile")
DEBUG: Checking if PR has been edited (branch="renovate/node-17.x-lockfile")
DEBUG: branch.isModified(): using cached result "false" (branch="renovate/node-17.x-lockfile")
DEBUG: Found existing branch PR (branch="renovate/node-17.x-lockfile")
DEBUG: Checking schedule(at any time, null) (branch="renovate/node-17.x-lockfile")
DEBUG: No schedule defined (branch="renovate/node-17.x-lockfile")
DEBUG: rebaseWhen=never so skipping branch update check (branch="renovate/node-17.x-lockfile")
DEBUG: getBranchPr(renovate/node-17.x-lockfile)
DEBUG: findPr(renovate/node-17.x-lockfile, undefined, open)
```

**Reply by rarkins** [reply]
Automerge is not compatible with rebaseWhen=never

**Reply by dave-yotta** [reply]
Thanks, that works!  I did spot mention under rebaseWhen "auto" that automerge will cause "behind-base-branch" to be used, but, there's nothing mentioned that anything else will cause it not to work :/  This branch was up to date; nothing to rebase.

I did find this too: https://docs.renovatebot.com/key-concepts/automerge/#frequent-problems-and-how-to-resolve-them might be worth adding!

Thanks again.
